ui/src/connection.ts

```diff
@@ -136,26 +137,50 @@ export function generateSecret(): string {
   return randomID(SECRET_LEN)
 }
 
-async function rawCommand<T>(baseUrl: string, command: unknown): Promise<T> {
-  return fetch(`${baseUrl}/command`, {
-    method: 'POST',
-    mode: 'same-origin',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(command),
-  }).then(res => {
-    if (res.status !== 200) {
-      throw new Error('Unexpected status code: ' + res.status);
+async function rawCommand<T>(baseUrl: string, command: unknown, attempt?: number): Promise<T> {
```

Comment:
I'm not a huge fan of leaking the `attempt` parameter to the API, even though it's probably not the end of the world. Actually, the whole retry logic could maybe be split in a higher order function?

Reply:
I would postpone this to the moment where it will actually bring value, e.g. when we need another retry logic. For now it's just good enough IMO